btp-document/Smart-Contracts-ICON.md

```diff
@@ -0,0 +1,387 @@
+## Smart Contracts on ICON Deployment
```

Comment:
For smart contract deployment (both for ICON and Moonbeam), it would be useful to have the error code reference somewhere, or at least a link pointing to files where those error codes are defined.

For example when I get a reverted transaction like the following, when calling ICON BMC, I'm not sure what to do with it

```
  "failure": {
    "code": "0x2e",
    "message": "Reverted(14)"
  },
```